integration/securitypolicy_test.go

```diff
@@ -0,0 +1,400 @@
+
+						policyOnTyk, err := klient.Universal.Portal().Policy().Get(reqCtx, policyOnK8s.Status.PolID)
+						eval.NoErr(err)
+
+						eval.True(policyOnK8s.Spec.MID == policyOnTyk.MID &&
+							len(policyOnK8s.Spec.Tags) == len(policyOnTyk.Tags) &&
+							len(policyOnTyk.Tags) == 1 &&
+							policyOnTyk.Tags[0] == initialK8sPolicyTag &&
+							policyOnTyk.Rate == initialK8sPolicyRate &&
+							policyOnTyk.State == initialK8sPolicyState)
+						return true
+					}),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				return ctx
+			}).
+		Assess("Create a drift between k8s and Dashboard",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				// In order to create a drift between k8s and Dashboard, first delete the SecurityPolicy controlled
+				// by Operator from Dashboard. Now, in the next reconciliation, Operator should recreate same
+				// SecurityPolicy on Tyk Side based on the SecurityPolicy CR on k8s.
+
+				eval := is.New(t)
+
+				// Delete an existing Policy from Dashboard to create drift between Tyk and K8s state.
+				err := klient.Universal.Portal().Policy().Delete(reqCtx, policyCR.Status.PolID)
+				eval.NoErr(err)
+
+				// Ensure that policy is deleted from Tyk.
+				_, err = klient.Universal.Portal().Policy().Get(reqCtx, policyCR.Status.PolID)
+				eval.True(tykClient.IsNotFound(err))
+
+				// After reconciliation, Operator should detect drift and recreate non-existing policy on Tyk side.
+				err = wait.For(func() (done bool, err error) {
+					_, err = polRec.Reconcile(ctx, ctrl.Request{NamespacedName: cr.ObjectKeyFromObject(&policyCR)})
+					return err == nil, err
+				})
+				eval.NoErr(err)
+
+				err = wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
+						policyOnK8s, ok := object.(*v1alpha1.SecurityPolicy)
+						eval.True(ok)
+
+						newSpec, err := klient.Universal.Portal().Policy().Get(reqCtx, policyOnK8s.Status.PolID)
+						eval.NoErr(err)
+
+						eval.True(hasSameValues(&policyOnK8s.Spec, newSpec, policyOnK8s.Status.PolID))
+						return true
+					}),
+					wait.WithTimeout(defaultWaitTimeout),
+					wait.WithInterval(defaultWaitInterval),
+				)
+				eval.NoErr(err)
+
+				return ctx
+			}).
+		Assess("Must be restored from Dashboard updates to K8s state on the next reconciliation",
+			func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+				// Assume that the user updated the Policy from Dashboard, which creates a drift between k8s state
+				// and Tyk. In order to establish consistency, Operator must update Security Policy based on the
+				// k8s state on next reconciliation - so that k8s remains as a source of truth.
+				eval := is.New(t)
+
+				err := wait.For(
+					conditions.New(c.Client().Resources()).ResourceMatch(&policyCR, func(object k8s.Object) bool {
```

Comment:
I think we don't need `ResourceMatch` here since we are not doing any operator on k8s resource. We are changing things on Tyk side. You can use `err = wait.For(func() (done bool, err error){}`

Reply:
You are right, we only run `DeepCopy` on resource's Spec. So, we do not need `conditions`. Updated accordingly 👍 thanks!